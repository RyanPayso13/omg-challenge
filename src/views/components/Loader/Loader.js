import React from 'react';
import { css } from '@emotion/core';
import { SyncLoader } from 'react-spinners';
import { OverlayWrap } from '../Styled';

const Loader = ({ isLoading }) => {
    
    const loader = css`
        align-self: center;
        margin: 0 auto; 
    `;

    return (
        <React.Fragment>
            {isLoading && 
                <OverlayWrap>
                    <SyncLoader
                        css={ loader }
                        color='#3C98EE' />
                </OverlayWrap>
            }
        </React.Fragment>
    );
};

export default Loader;