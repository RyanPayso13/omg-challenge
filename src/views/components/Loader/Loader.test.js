import React from 'react';
import { render } from '@testing-library/react';
import Loader from './Loader.js';

describe('<Loader />', () => {

    it('should render', () => {
        const { container } = render(<Loader isLoading={ true } />);
        expect(container.firstChild).toMatchSnapshot();    
    });

    it('should not render', () => {
        const { container } = render(<Loader isLoading={ false } />);
        expect(container.firstChild).toBeNull();    
    });

});
