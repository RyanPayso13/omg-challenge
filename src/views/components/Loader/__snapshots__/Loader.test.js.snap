// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(256,256,256,.9);
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  class="c0"
>
  <div
    class="css-15yujpl-Loader"
  >
    <div
      class="css-1v2vqtz-style"
    />
    <div
      class="css-3imrv7-style"
    />
    <div
      class="css-9tspmy-style"
    />
  </div>
</div>
`;
